segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -223,23 +237,32 @@ public void setupAppend(SetupAppend setupAppend) {
                         });
     }
 
-    private void populateEventSizeForIndexSegmentAppends(UUID writer, String indexSegment, long requestId) {
+    private void createIndexSegmentIfNotExists(final UUID writer, final String indexSegment, long requestId) {
         store.getStreamSegmentInfo(indexSegment, TIMEOUT)
                 .whenComplete((properties, u) -> {
-                    if (u != null) {
-                        u = Exceptions.unwrap(u);
-                        if (u instanceof NoSuchSegmentException || u instanceof StreamSegmentNotExistsException) {
-                            log.warn("could not find the segment {}", indexSegment);
-                        } else {
-                            handleException(writer, requestId, indexSegment, "populating event size for index appends", u);
-                        }
-                    } else {
-                        if (properties != null) {
-                            Map<AttributeId, Long> attributes =  properties.getAttributes();
-                            this.writerStates.put(Pair.of(indexSegment, writer), new WriterState(attributes.get(Attributes.ALLOWED_INDEX_SEG_EVENT_SIZE)));
+                    try {
+                        if (u != null) {
+                            u = Exceptions.unwrap(u);
+                            if (u instanceof NoSuchSegmentException || u instanceof StreamSegmentNotExistsException) {
+                                log.info("Creating index segment {}.", indexSegment);
+                                Collection<AttributeUpdate> attributes = Arrays.asList(
+                                        new AttributeUpdate(CREATION_TIME, AttributeUpdateType.None, System.currentTimeMillis()),
+                                        new AttributeUpdate(ATTRIBUTE_SEGMENT_TYPE, AttributeUpdateType.None, SegmentType.STREAM_SEGMENT.getValue()),
+                                        new AttributeUpdate(ALLOWED_INDEX_SEG_EVENT_SIZE, AttributeUpdateType.None, 24L)
+                                );
+                                store.createStreamSegment(indexSegment, SegmentType.STREAM_SEGMENT, attributes, TIMEOUT).join();
+                                this.writerStates.put(Pair.of(indexSegment, writer), new WriterState(24L));
+                            } else {
+                                throw u;
+                            }
                         } else {
-                            log.debug("could not get properties for a given segment {}", indexSegment);
+                            if (properties != null) {
+                                Map<AttributeId, Long> attributes = properties.getAttributes();
+                                this.writerStates.put(Pair.of(indexSegment, writer), new WriterState(attributes.get(Attributes.ALLOWED_INDEX_SEG_EVENT_SIZE)));
```

Comment:
This PR is very confused as to what the argument to `new WriterState(` is. In some places it is acting as event number, in others as event count, and elsewhere it is passes as `24` as though it were the size of the entries in the index.